conda_smithy/lint_recipe.py

```diff
@@ -1017,7 +1028,383 @@ def sort_osx(versions):
     return lints, hints
 
 
-def run_conda_forge_specific(meta, recipe_dir, lints, hints):
+def lintify_recipe_yaml(
```

Comment:
@beckermr is that then opposing isuruf's previous comments about making an external package for `rattler-build-conda-compat`? Because this was on specific request. I think we'd be fine for moving the functionality back into smithy.

Reply:
That request pertained only to the APIs in conda-build, not to the stuff smithy handles. The request as I understood it was to mirror the split in apis between conda-build and conda-smithy.

Reply:
hey @beckermr ! 
I will move linting checks back into conda-smithy as I agree with you that it will give more control for conda-forge team.
the schema lint will be in rattler_build_conda_compat

Reply:
@beckermr 
I've moved all rattler lint checks in `conda_smithy/rattler_linter.py`. Let me know please what do you think about this approach. Thanks